doc/source/parameters/cfd/linear_solver_control.rst

```diff
@@ -0,0 +1,157 @@
+
+	Typical values of the ``minimum residual`` for simple simulations can be quite low, for instance ``set minimum residual = 1e-13``. However, such a tolerance can be impossible to attain for more complex simulations, and so the ``minimum residual`` can be increased, for instance ``set minimum residual = 1e-5``.
+
+* The ``relative residual`` weights the :math:`\mathcal{L}_2`-norm of the system RHS. This product is then compared to the ``minimum residual`` to assess for convergence of the iteration.
+.. tip::
+	The default value of ``1e-3`` is a good start value. To lower the computational cost for more complex simulations, it can be lowered to ``set relative residual = 1e-4``.
+
+* The ``max iters`` puts a hard stop on the number of solver iterations (number of steps printed when ``set verbosity = verbose``).
+.. tip::
+	If ``max iters`` is reached, the code will throw this type of message: 
+	
+	.. code-block:: text
+	
+		GMRES solver failed! Trying with a higher preconditioner fill level.
+
+	meaning that the code increases the preconditioner fill (see definition below) in order to converge within the number of solver iterations. If you encounter this, consider increasing the ``max iters``.
+
+* ``force linear solver continuation`` enables, when set to ``true``, to force the linear solver to continue, even if the ``minimum residual`` is not reached. Only available for ``GMRES`` solver within the ``gls_navier_stokes`` application.
+.. warning::
+	With this mode on, errors on the linear solver convergence are not thrown. Forcing the solver to continue can be useful for debugging purposes if a given iteration is hard to pass, but use with caution!
+
+* ``max krylov vectors`` sets the maximum number of krylov vectors for ``GMRES`` and ``AMG`` solvers.
+.. tip::
+	Consider using ``set max krylov vectors = 200`` for complex simulations with convergence issues. 
+
+* ``ilu preconditioner fill``, ``ilu preconditioner absolute tolerance`` and ``ilu preconditioner relative tolerance`` control the ILU preconditioner for ``method`` using ILU preconditioner (``gmres`` and ``bicgstab``). Conversely, ``amg preconditioner ilu fill``, ``amg preconditioner ilu absolute tolerance`` and ``amg preconditioner ilu relative tolerance`` control the ILU coarsener and smoother for the AMG preconditioner.
+ 
+.. tip::
+	The default values for these parameters are good start values. 
+
+	For each iteration of the linear solver (at the beginning of which the tolerance of the iterative solver is computed, as printed if ``set verbosity = verbose``), the chosen solver starts by using the ``preconditioner fill`` given in the parameter file. If for any reason the linear solver would have crashed, it will restart with a fill level increased by 1. This restart process will happen up to a maximum of 20 times, after which it will let the solver crash. 
```

Comment:
It does not happen up to 20 times, I think the max is like 3 now?
@Luckabarbeau  you know the exact value?

Reply:
Oh yes it seems in gls_navier_stokes that now it is 3 max! We will also need to update the comments in the code about this ^^"